solr/CHANGES.txt

```diff
@@ -243,6 +243,8 @@ Bug Fixes
 * SOLR-17692: Core unload/deletion now preempts all forms of ongoing "recovery", rather than inadvertently waiting for
   completion in some cases. (Jason Gerlowski)
 
+* SOLR-17294: The stall detection in the ConcurrentUpdateSolrClients easily detects false positives. (Mark Miller, Jason Gerlowski)
```

Comment:
[-0] The user-facing benefit might be clearer if this is phrased in terms of the new/improved behavior. e.g.

> ConcurrentUpdateSolrClient no longer detects "false-positive" stalls, improving throughput and decreasing logging "noise" (Mark Miller)

Reply:
> improving throughput and decreasing logging "noise"

I'd typically be fine with that, but here it requires saying more than I'm comfortable with. In the indexing I was recently doing, it completely hosed my indexing job and I had to work around it to even finish indexing. Which I guess you could argue is still a loss of throughput (infinite loss?), but I'm not confident saying that it just increases throughput. Similar with logging "noise". Noise to me is unnecessary logging, but this logging is coming from a false claim that there is a stall, so reduced logging noise is also accurate I suppose, but feels misleading.

The positive is much harder to define exactly, whereas the problem, it was easy to detect false stalls, that I'm sure of.